cancermuts_streamlit/streamlit_utils.py

```diff
@@ -0,0 +1,80 @@
+import streamlit as st
+import os
+import pandas as pd
+
+def filter_dataframe(df: pd.DataFrame) -> pd.DataFrame:
+    addFilter = st.checkbox("Add filter")
+
+    if not addFilter:
+        return df
+    
+    df = df.copy()
+
+    for col in df.columns:
+        try:
+            df[col] = pd.to_datetime(df[col], format='%d-%m-%Y')
+        except Exception:
+            pass
+
+    filter_container = st.container()
+    with filter_container:
+        filter_columns = st.multiselect("Filter dataframe on", df.columns)
+
+    for col in filter_columns:
+        left, right = st.columns((1, 20))
+        left.write("↳")
+
+        if df[col].dtype == 'datetime64[ns]':
+        #if len(df[col]) > 0 and isinstance(df[col][0], pd._libs.tslibs.timestamps.Timestamp):
+            user_date_input = right.date_input(f"Values for {col}",
+            value=(
+                df[col].min(),
+                df[col].max(),
+                ),)
+            
+            if len(user_date_input) == 2:
+                user_date_input = tuple(map(pd.to_datetime, user_date_input))
+                start_date, end_date = user_date_input
+                df = df.loc[df[col].between(start_date, end_date)]
+
+        else:
+            user_text_input = right.selectbox(
+                f"{col}", df[col])
+            #if user_text_input:
+            df = df[df[col].str.contains(user_text_input)]
+
+    for col in df.columns:
+        if df[col].dtype == 'datetime64[ns]':
+            df[col] = df[col].dt.strftime("%d-%m-%Y")
+
+    return df
+
+def get_selection(df: pd.DataFrame) -> pd.DataFrame:
+    df_with_selections = df.copy()
+    df_with_selections.insert(0, "Select", False)
+
+    edited_df = st.data_editor(
+        df_with_selections,
+        hide_index=True,
+        column_config={"Select": st.column_config.CheckboxColumn(required=True)},
+        disabled=df.columns,
+    )
+    # Filter the dataframe using the temporary column, then drop the column
+    selected_rows = edited_df[edited_df.Select]
+    return selected_rows.drop('Select', axis=1)
+
+def get_database_dir(var_name='MAVISP_DATABASE', default_dir='./database'):
+    dir_name = os.getenv(var_name)
+    if dir_name is None:
+        return default_dir
+    return var_name
+
+def get_database_dir(database):
+    return os.path.join(os.getcwd(), database)

```

Comment:
double definition - I assume the first one was just copied from MAVISp

Reply:
Fixed